evap/static/js/datagrid.js

```diff
@@ -0,0 +1,314 @@
+class DataGrid {
+    // Use init instead of constructor to enable access to this before calling super
+    init({storageKey, container, searchInput}) {
+        this.storageKey = storageKey;
+        this.container = container;
+        this.sortableColumns = Array.from(this.sortableHeaders.keys());
+        this.searchInput = searchInput;
+        this.rows = this.fetchRowData();
+        this.loadFromStorage();
+        this.bindEvents();
+    }
+
+    bindEvents() {
+        this.delayTimer = null;
+        this.searchInput.on("change paste input", () => {
+            clearTimeout(this.delayTimer);
+            this.delayTimer = setTimeout(() => {
+                this.search(this.searchInput.val());
+            }, 200);
+        }).keypress(event => {
+            // after enter, unfocus the search input to collapse the screen keyboard
+            if (event.keyCode === 13) {
+                event.target.blur();
+            }
+        });
+
+        for (const [column, header] of this.sortableHeaders) {
+            header.click(() => {
+                const ordering = header.hasClass("col-sorting-asc") ? "desc" : "asc";
+                this.sort([[column, ordering]]);
+            });
+        }
+    }
+
+    fetchRowData() {
+        return this.container.children().get()
+            .map(row => {
+                let data = {
+                    element: row,
+                    orderValues: new Map(),
+                    filterValues: new Map(),
+                };
+                data.searchWords = this.findSearchableCells($(row))
```

Comment:
why not put that into the initialization above?

Reply:
Primarly to reduce bloat in the object literal. I now set a variable before it, so you see all keys in the object literal